agent/consul/config_endpoint.go

```diff
@@ -513,11 +510,12 @@ func (c *ConfigEntry) ResolveServiceConfig(args *structs.ServiceConfigRequest, r
 				ranOnce = true
 			}
 
```

Comment:
@mkeeler I did not move this entire piece of code to be re-usable and instead picked out relevant calls and moved it into `merge_service_config.go/mergeNodeServiceWithCentralConfig`.
The main reason being there is the optimization here for spurious wakeup suppression. 

Do we want to add this optimization in some manner on looping through all service resolutions?

Reply:
Would it be terribly difficult to bring over the spurious wakeup suppression. While this should just be the UI performing these queries we have the potential to be constantly sending back this response if we don't have the suppression.